user-service/src/main/java/com/movie/recommender/user/controller/UserController.java

```diff
@@ -0,0 +1,46 @@
+package com.movie.recommender.user.controller;
+
+import com.movie.recommender.user.model.dto.UserDTO;
+import com.movie.recommender.user.model.entity.User;
+import com.movie.recommender.user.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private AuthenticationManager authenticationManager;
+
+    // Регистрация пользователя через DTO
+    @PostMapping("/register")
+    public ResponseEntity<UserDTO> registerUser(@RequestBody UserDTO userDTO) {
+        UserDTO registeredUser = userService.registerUser(userDTO);
+        return new ResponseEntity<>(registeredUser, HttpStatus.CREATED);
+    }
+
+    @PostMapping("/login")
+    public ResponseEntity<String> loginUser(@RequestBody UserDTO userDTO) {
+        try {
```

Comment:
add a global exception handler, similar to the one in `location service`
there may be shared logic between the exception handlers, so consider moving this logic to a `common` module to maintain consistency and reusability across the project